### PR TITLE
fix: Change default SECRET_KEY, improve docs and banner warning on de…

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY
Simple change to better alert our users when the default `SECRET_KEY` has not been changed.

When the default SECRET_KEY is detected, superset will log:
```
superset_app             | --------------------------------------------------------------------------------
superset_app             |                                     WARNING
superset_app             | --------------------------------------------------------------------------------
superset_app             | A Default SECRET_KEY was detected please use superset_config.py to override it.
superset_app             | Use a strong complex alphanumeric string and use a tool to help you generate
superset_app             | a sufficiently random sequence, ex: openssl rand -base64 42
superset_app             | --------------------------------------------------------------------------------
superset_app             | --------------------------------------------------------------------------------
```

The default `SECRET_KEY` changes to a more standard default, making it more obvious that a change is needed also.
This can break current installations that were relying on the previous default `SECRET_KEY` I believe this is reasonable, since this is a dangerous setting and a proper setting is imperative. 

This PR also adds a cli command to help user's migrate their current secrets. They have to set `PREVIOUS_SECRET_KEY` with their old SECRET and set their new secret with `SECRET_KEY` then run `superset re-encrypt-secrets`. Props to @craig-rueda since this script is highly based (adapted) to some work he has done.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by betodealmeida — APPROVED** [review]
This is awesome! Bonus points for telling the user how to generate the key in a safe way. :)

**Comment by dpgaspar** [comment]
> This is awesome! Bonus points for telling the user how to generate the key in a safe way. :)

I'm proposing using `openssl rand -base64 42` on the message, are you thinking about something else?

File: superset/initialization/__init__.py [hunk 1]
```diff
@@ -572,12 +573,27 @@ def init_app_in_ctx(self) -> None:
...
+            bottom_banner = 80 * "-" + "\n" + 80 * "-"
+            logger.warning(top_banner)
+            logger.warning(
+                "A Default SECRET_KEY was detected please use superset_config.py "
+                "to override it.\n"
+                "Use a strong complex alphanumeric string and use a tool to help"
+                " you generate \n"
+                "a sufficiently random sequence, ex: openssl rand -base64 42"
+            )
+            logger.warning(bottom_banner)
```
Comment (etr2460) [inline superset/initialization/__init__.py:588]: If this happens should we do a `sys.exit` call as well so the server doesn't even start up?

**Review by craig-rueda — APPROVED** [review]

File: superset/initialization/__init__.py [hunk 2]
```diff
@@ -572,12 +573,27 @@ def init_app_in_ctx(self) -> None:
 
         self.init_views()
 
+    def check_secret_key(self) -> None:
+        if self.config["SECRET_KEY"] == CHANGE_ME_SECRET_KEY:
+            top_banner = 80 * "-" + "\n" + 36 * " " + "WARNING\n" + 80 * "-"
+            bottom_banner = 80 * "-" + "\n" + 80 * "-"
+            logger.warning(top_banner)
+            logger.warning(
+                "A Default SECRET_KEY was detected please use superset_config.py "
```
Comment (villebro) [inline superset/initialization/__init__.py:582]: nit:
```suggestion
                "A Default SECRET_KEY was detected, please use superset_config.py "
```

**Review by villebro — COMMENTED** [review]
Change LGTM with minor nit. However, since this will break every single installation that is currently using the default key, I would prefer to either make `CHANGE_ME_SECRET_KEY` equal to the old default `SECRET_KEY`, or add a note to `UPDATING.md`. Otherwise I fear we will be swampled by questions on both GitHub and Slack from users wondering why their devenvs on master (hopefully not prod envs!) have stopped working.

File: superset/initialization/__init__.py (hunk 1 above)
Comment (dpgaspar) [reply superset/initialization/__init__.py:588]: I evaluated something similar, but I'm avoiding break pip installations and docker. It's really nice to have a really fast frictionless first installations.

**Comment by dpgaspar** [comment]
> Change LGTM with minor nit. However, since this will break every single installation that is currently using the default key, I would prefer to either make `CHANGE_ME_SECRET_KEY` equal to the old default `SECRET_KEY`, or add a note to `UPDATING.md`. Otherwise I fear we will be swampled by questions on both GitHub and Slack from users wondering why their devenvs on master (hopefully not prod envs!) have stopped working.

**Review by villebro — APPROVED** [review]
Thanks for adding the rotation script, looks awesome!

**Comment by michael-s-molina** [comment]
Love the instructions on how to generate a new key 👏🏼

**Pull request merged**